chips/rp2040/src/pwm.rs

```diff
@@ -0,0 +1,1290 @@
+    //
+    // The counter must be running. The method will return once the retardation
+    // is complete.
+    fn retard_count(&self, channel_number: ChannelNumber) {
+        self.registers.ch[channel_number as usize]
+            .csr
+            .modify(CSR::PH_RET::SET);
+        while self.registers.ch[channel_number as usize]
+            .csr
+            .read(CSR::PH_RET)
+            == 1
+        {}
+    }
+
+    // Enable interrupt on the given PWM channel
+    fn enable_interrupt(&self, channel_number: ChannelNumber) {
+        // What about adding a new method to the register interface which performs
+        // a bitwise OR and another one for AND?
+        let mask = self.registers.inte.read(CH::CH);
+        self.registers
+            .inte
+            .modify(CH::CH.val(mask | 1 << channel_number as u32));
+    }
+
+    // Disable interrupt on the given PWM channel
+    fn disable_interrupt(&self, channel_number: ChannelNumber) {
+        let mask = self.registers.inte.read(CH::CH);
+        self.registers
+            .inte
+            .modify(CH::CH.val(mask & !(1 << channel_number as u32)));
+    }
+
+    // Enable multiple channel interrupts at once.
+    //
+    // Bits 0 to 7 ==> enable channel 0-7 interrupts.
+    fn enable_mask_interrupt(&self, mask: u8) {
+        let old_mask = self.registers.inte.read(CH::CH);
+        self.registers
+            .inte
+            .modify(CH::CH.val(old_mask | mask as u32));
+    }
+
+    // Disable multiple channel interrupts at once.
+    //
+    // Bits 0 to 7 ==> disable channel 0-7 interrupts.
+    fn disable_mask_interrupt(&self, mask: u8) {
+        let old_mask = self.registers.inte.read(CH::CH);
+        self.registers
+            .inte
+            .modify(CH::CH.val(old_mask & !mask as u32));
+    }
+
+    // Clear interrupt flag
+    fn clear_interrupt(&self, channel_number: ChannelNumber) {
+        self.registers
+            .intr
+            .write(CH::CH.val(1 << channel_number as u32));
+    }
+
+    // Force interrupt on the given channel
+    fn force_interrupt(&self, channel_number: ChannelNumber) {
+        let mask = self.registers.intf.read(CH::CH);
+        self.registers
+            .intf
+            .modify(CH::CH.val(mask | 1 << channel_number as u32));
+    }
+
+    // Unforce interrupt
+    fn unforce_interrupt(&self, channel_number: ChannelNumber) {
+        let mask = self.registers.intf.read(CH::CH);
+        self.registers
+            .intf
+            .modify(CH::CH.val(mask & !(1 << channel_number as u32)));
+    }
+
+    // Get interrupt status
+    fn get_interrupt_status(&self, channel_number: ChannelNumber) -> bool {
+        (self.registers.ints.read(CH::CH) & 1 << channel_number as u32) != 0
+    }
+
+    // Handle PWM interrupts
+    //
+    // This method should be called only inside crate::chip::Rp2040DefaultPeripherals::service_interrupt.
+    pub fn handle_interrupt(&self) {
```

Comment:
```suggestion
    pub(crate) fn handle_interrupt(&self) {
```

Reply:
As this is not through a HIL, I suggest limiting it to the crate.

Reply:
Removed it since the HIL provides no support for interrupts.